examples/input-element-default-checkedness/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+use percy_dom::{event::VirtualEvents, html, JsCast, VirtualNode};
+use wasm_bindgen_test::*;
+use web_sys::{Element, HtmlInputElement, Node};
+
+wasm_bindgen_test_configure!(run_in_browser);
+
+fn create_my_default_checked_checkbox() -> HtmlInputElement {
+    let document = web_sys::window().unwrap().document().unwrap();
+
+    let checkbox = document
+        .create_element("input")
+        .unwrap()
+        .dyn_into::<HtmlInputElement>()
+        .unwrap();
+    checkbox.set_id("my_default_checked_checkbox");
+    checkbox.set_type("checkbox");
+    checkbox.set_default_checked(true);
+    checkbox.set_checked(true);
+
+    checkbox
+}
+
+fn setup_percy_dom_with_appended_child_checkbox(
+) -> (VirtualNode, Node, VirtualEvents, HtmlInputElement) {
+    let my_default_checked_checkbox = create_my_default_checked_checkbox();
+    let my_default_checked_checkbox_append = my_default_checked_checkbox.clone();
+
+    // This will be a checkbox that `percy-dom` controls.
+    let vdom_root_node = html! {
+        <div id="root">
+            <div
+                id="checkbox_holder"
+                key="key"
+                on_create_element=move |elem: Element| {
+                    elem.append_child(&my_default_checked_checkbox_append).unwrap();
+                }
+            >
+                <input id="percy_checkbox" type="checkbox" checked=true>
+            </div>
+        </div>
+    };
+
+    // Create the DOM nodes from the virtual DOM.
+    let mut events = VirtualEvents::new();
+    let (root_node, event_node) = vdom_root_node.create_dom_node(&mut events);
+    events.set_root(event_node);
+
+    (
+        vdom_root_node,
+        root_node,
+        events,
+        my_default_checked_checkbox,
+    )
+}
+
+fn uncheck_both_checkboxes(
+    vdom_root: &VirtualNode,
+    dom_root: Node,
+    events: &mut VirtualEvents,
+    my_default_checked_checkbox: HtmlInputElement,
+) {
+    // Update the default-checked checkbox we maintain to be unchecked
+    my_default_checked_checkbox.set_checked(false);
+
+    // Update the percy checkbox to unchecked
+    let new_vdom_root = html! {
+        <div id="root">
+            <div id="checkbox_holder">
+                <input id="percy_checkbox" type="checkbox" checked=false>
+            </div>
+        </div>
+    };
+    let patches = percy_dom::diff(&vdom_root, &new_vdom_root);
+    percy_dom::patch(dom_root.clone(), &new_vdom_root, events, &patches).unwrap();
+}
+
+/// Verify that the `percy-dom`-controlled checkbox, `percy_checkbox`, and the
+/// `my_default_checked_checkbox` behave exactly as we expect when updating the checkedness:
+/// - Both `percy_checkbox`'s checkedness and default checkedness should change.
+/// - Only `my_default_checked_checkbox`'s checkedness should change, not default checkedness.
+#[wasm_bindgen_test]
+fn checkbox_checkedness_update_test() {
```

Comment:
File looks well structured so I didn't read these latest revisions.

One request: move this function to the top.

This lets the user immediately see where the runtime logic starts and step through it themselves.

vs. seeing a pile of functions before they finally get here and start to build a mental map of what this example is doing.

After that, this can land as is. I didn't read it, but that's mainly because it looks so well structured that if I have any feedback it'll be easy to just apply myself in the future.

Reply:
Deleted the whole thing as it's a bad example now that a dev doesn't need to go to such lengths to manipulate default checkedness. It's in the commit history if parts of it would be useful later.